plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -183,9 +197,23 @@ private static void generateEnumAttributeSchema(JSONObject attributeSchemaTempla
             }
             attributeSchemaTemplate.put(attribute.getName(),
                 new JSONObject()
+                    .put("description", retrieveDocStringFromAttribute(baseConfigurator.getTarget(), attribute.name))
                     .put("type", "string")
                     .put("enum", new JSONArray(attributeList)));
         }
     }
+
+    public static String retrieveDocStringFromAttribute(Class baseConfigClass, String attributeName) {
+        String htmlDocString = null;
+        try {
+            htmlDocString = ConfigurationAsCode.get().getHtmlHelp(baseConfigClass, attributeName);
+        } catch (IOException e) {
+            LOGGER.warning("Error getting help document for attribute : " + e);
+        }
+        if(htmlDocString!=null) {
```

Comment:
the above method never returns null afaik? unless there was an exception, but anyway, I would move this up into the try block and then return the variable directly there